server/src/main/java/org/elasticsearch/telemetry/metric/MeterService.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.telemetry.metric;
+
+public interface MeterService {
+    void setEnabled(boolean enabled);
```

Comment:
It seems like this is only made public so that tests can set it? I suggest having it be package private, and using a test only adaptor (class in the same package) to allow setting it for tests.

Reply:
Removed interface and made it package private in `APMMeterService`.  Added  `TestAPMMeterService` to expose the method in unit tests.